src/main/java/lu/oop/server/api/controllers/ComplaintController.java

```diff
@@ -87,8 +116,22 @@ public ResponseEntity<IComplaintModel> assignComplaint(@PathVariable int id, @Pa
   @PreAuthorize("hasRole('admin')")
   public ResponseEntity<IComplaintModel> closeComplaint(@PathVariable int id){
     try {
+
+      IAdminModel loggedInUser = (IAdminModel) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+      Optional<IUserModel> user = userService.getById(loggedInUser.getId());
+      IAdminModel admin = (IAdminModel) user.get();
       Long complaintId = Long.valueOf(id);
 
+      if(!admin.getAssignedComplaints()
+      .stream()
+      .map(ComplaintModel::getId)
+      .anyMatch(value -> value.equals(complaintId))) 
+      {
+          // Can only resolve assigned complaint
+          logger.warn("Admin attempted to close not assigned complaint");
```

Comment:
The logs are good. Will help the client team to understand what is happening. But there is also an error class which standardizes the responses